File: frontend/app/lib/api/types/library.ts
export interface ResumeListItem {
	id: string;
	name: string;
	vacancyTitle: string | null;
	language: string;
	createdAt: string;
	updatedAt: string;
}

export interface ResumeDocument extends ResumeListItem {
	html: string;
}

export interface CoverLetterListItem {
	id: string;
	name: string;
	vacancyTitle: string | null;
	createdAt: string;
	updatedAt: string;
}

export interface CoverLetterDocument extends CoverLetterListItem {
	html: string;
}

export interface UpdateLibraryDocumentInput {
	id: string;
	html: string;
}

File: backend/app/schemas/library.py
import datetime
import uuid

from schemas import CamelModel


class DocumentHtmlUpdate(CamelModel):
    """Payload used to replace a saved document's HTML."""

    html: str


class ResumeListItem(CamelModel):
    """A saved resume as shown in the created-documents library."""

    id: uuid.UUID
    name: str
    vacancy_title: str | None
    language: str
    created_at: datetime.datetime
    updated_at: datetime.datetime


class ResumeDetail(ResumeListItem):
    """A saved resume including its rendered HTML for previewing."""

    html: str


class CoverLetterListItem(CamelModel):
    """A saved cover letter as shown in the created-documents library."""

    id: uuid.UUID
    name: str
    vacancy_title: str | None
    created_at: datetime.datetime
    updated_at: datetime.datetime


class CoverLetterDetail(CoverLetterListItem):
    """A saved cover letter including its rendered HTML for previewing."""

    html: str
